src/serve/search.rs

```diff
@@ -0,0 +1,435 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Search functionality for the semantic convention registry.
+
+use std::sync::Arc;
+
+use weaver_forge::v2::{
+    attribute::Attribute, entity::Entity, event::Event, metric::Metric,
+    registry::ForgeResolvedRegistry, span::Span,
+};
+use weaver_semconv::stability::Stability;
+
+use super::types::{ScoredResult, SearchResult, SearchType};
+
+/// Search context for performing fuzzy searches across the registry.
+pub struct SearchContext {
```

Comment:
This is an idea for the future, but similar to how some UIs work or JAQ, we may want to turn this into a:

```
pub struct SearchContext<'a> {
  items: Vec<SearchableItem<'a>>
}
impl <'a> SearchContext<'a> {
  fn from_registry(registry: &'a ForgeRegistry) -> SearchContext<'a> { ... }
}
```

This could help reduce the memory overhead of copying all the data for search if we hit large / many repositories.  We just need to make sure the out registry's lifetime is protected when we create the SearchContext so lifetime magic all works together.